pkg/placement/placement-resolver.go

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright 2023 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package placement
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/kubestellar/kubestellar/api/edge/v1alpha1"
+	"github.com/kubestellar/kubestellar/pkg/util"
+)
+
+// PlacementResolver abstracts the functionality needed from a placement resolver,
+// to simplify its usage and keep implementation details inaccessible.
+type PlacementResolver interface {
+	// GeneratePlacementDecision returns the placement decision for the given
+	// placement key. If a key does not exist in the cache, it is created.
+	GeneratePlacementDecision(placementKey string) (*v1alpha1.PlacementDecisionSpec, error)
+	// ComparePlacementDecision compares the given placement decision spec
+	// with the maintained placement decision for the given placement key.
+	ComparePlacementDecision(placementKey string,
+		placementDecisionSpec *v1alpha1.PlacementDecisionSpec) bool
+	// UpdateObjects updates the maintained placement's
+	// objects data for the given placement key.
+	// If a key does not exist in the cache, it is created.
+	// The returned bool indicates whether the placement resolution was changed.
+	UpdateObjects(placementKey string, obj runtime.Object) (bool, error)
```

Comment:
The name of this method uses the plural "objects" but the method actually takes only a single object.

The word "update" is very general and vague. I think it would be helpful to be clearer, saying instead that this is an "add".

Reply:
NoteObject was chosen.